assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js

```diff
@@ -78,46 +75,42 @@ function Media3pPane(props) {
     [insertElement]
   );
 
-  const { searchTerm, setSelectedProvider, setSearchTerm } = useMedia3p(
-    ({ state, actions }) => ({
-      searchTerm: state.searchTerm,
-      setSelectedProvider: actions.setSelectedProvider,
-      setSearchTerm: actions.setSearchTerm,
+  const {
+    searchTerm,
+    setSelectedProvider,
+    setSearchTerm,
+    unsplash,
+    selectedProviderState,
+  } = useMedia(
+    ({
+      media3p: {
+        state: { selectedProvider, searchTerm },
+        actions: { setSelectedProvider, setSearchTerm },
+        unsplash,
+      },
+      media3p,
```

Comment:
Does js not allow destructuring here, like 
media3p[selectedProvider ?? ProviderType.UNSPLASH]: {
  //needed state
}
?

Reply:
This isn't possible here, as 'selectedProvider' is not yet declared.